Refactor Interactions to enable loading historical Interactions

mnzaki opened the pull request:
`agent.processJWT` is now responsible for ~validating incoming JWTs,~ storing JWTs as necessary, and "encountering" new DIDs based on `jwt.payload.pca`
`Interaction.processInteractionToken` need not concern itself with those, and can remain rather "pure"

This PR does not affect the public API's inputs/outputs, but does affect behavior: if someone was directly using `Interaction.processInteractionToken` then they should now also handle the storage ~and validation~ concerns

chunningham reviewed (commented):
this solution is appropriate for the issue using only SDK changes, although IMO a better solution would be to adjust `validateJWT` in the library to either take options or remove time validation, which would let us keep `Interaction` as the security validation layer although we'd still have to come up with a neat way to validate (or not, if it's a replay) expiry time. Will approve if we need this ASAP

mnzaki commented:
So I haven't really arrived to a good answer to "should we be validating at this layer" and instead now split out `processInteractionToken` into another private method `_processToken` which is used internally when we don't want to validate.

Note that I still left the storage and pca processing changes (i.e. moved them to `agent.processJWT`) because I think those make a lot more sense there, and it improves performance quite a bit for loading stuff from DB

chunningham reviewed (approved):
nice, strong agree with what you've left in `Agent`. LGTM and we can plan further refactoring from here :+1:
